Resources/public/js/views/serverside/ez-roleserversideview.js

```diff
@@ -37,27 +43,134 @@ YUI.add('ez-roleserversideview', function (Y) {
+                    sectionName: sectionSelector.get('options').item(sectionSelectedIndex).get('text')
+                },
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
+                    roleName: button.getAttribute('data-role-name'),
+                    afterUpdateCallback: unsetLoading,
+                    limitationType: 'Section',
+                    section: section,
+                    sectionId: sectionSelector.get('options').item(sectionSelectedIndex).get('value'),
+                    sectionRestId: sectionSelector.get('options').item(sectionSelectedIndex).getAttribute('data-section-rest-id'),
+                    sectionName: sectionSelector.get('options').item(sectionSelectedIndex).get('text'),
+                };
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this._fireContentDiscover(button, unsetLoading, udwConfigData);
+        },
+
+        /**
+         * tap event handler on the role assign buttons. It launches the
+         * universal discovery widget so that the user can pick some contents.
          *
          * @method _pickSubtree
          * @protected
          * @param {EventFacade} e
          */
         _pickSubtree: function (e) {
             var button = e.target,
-                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button);
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                udwConfigData = {
+                    roleId: button.getAttribute('data-role-rest-id'),
+                    roleName: button.getAttribute('data-role-name'),
+                    afterUpdateCallback: unsetLoading,
+                };
 
             e.preventDefault();
             this._uiSetUDWButtonLoading(button);
+            this._fireContentDiscover(button, unsetLoading, udwConfigData);
+        },
+
+        /**
+         * tap event handler for assigning role with subtree limitation. It launches the
+         * universal discovery widget a first time so that the user can pick a location.
+         *
+         * @method _pickLocationLimitation
+         * @protected
+         * @param {EventFacade} e
+         */
+        _pickSubtreeLimitation: function (e) {
+            var button = e.target,
+                unsetLoading = Y.bind(this._uiUnsetUDWButtonLoading, this, button),
+                that = this;
+
+            e.preventDefault();
+            this._uiSetUDWButtonLoading(button);
+            this.fire('contentDiscover', {
+                config: {
+                    title: button.getAttribute('data-universaldiscovery-limit-subtree-title'),
+                    cancelDiscoverHandler: unsetLoading,
+                    multiple: true,
+                    contentDiscoveredHandler: function(e) {
+                        that._setSubtreeLimitation(button, this, e);
```

Comment:
I need to keep the udwView here (-> this), so I can't use Y.bind or I will loose the context of the udwView for the parameters of . _setSubtreeLimitation